minio/resource_minio_iam_group_policy_attachment.go

```diff
@@ -58,9 +58,10 @@ func minioReadGroupPolicyAttachment(ctx context.Context, d *schema.ResourceData,
 	minioAdmin := meta.(*S3MinioClient).S3Admin
 
 	var groupName = d.Get("group_name").(string)
+	var isLDAPGroup = LDAPGroupDistinguishedNamePattern.MatchString(groupName)
 
 	groupInfo, errGroup := minioAdmin.GetGroupDescription(ctx, groupName)
-	if errGroup != nil {
+	if errGroup != nil && !isLDAPGroup {
```

Comment:
I haven't tried it and don't have a docker-compose setup for a LDAP based minio.  So I'm guessing that when using LDAP that `GetGroupDescription` will return an error?

That seems slightly odd..  If it is that way, maybe a better way would be to simply not execute the call if it is an ldap group?

Reply:
Nope it does work just fine if the group actually exists. Here is an example:

```
$ mc admin group info myinstance "CN=my,OU=group,DC=my,DC=org"
Group: CN=my,OU=group,DC=my,DC=orgStatus: 
Policy: 
Members: 
```

Removing this would mean not being able to perform drift detection which would be very problematic IMHO

Reply:
Thank you for the detailed answer!  Trying to wrap my head around it.

If the call to `GetGroupDescription` should work, then I'm unsure what that `if` actually does.

* if the call to `GetGroupDescription` failed AND it's not a normal user THEN return an error from `minioReadGroupPolicyAttachment`

So it ignores failures getting the group description if the group is from LDAP.  Why can we ignore failures at this point?

Reply:
The reason for doing this is that `GetGroupDescription` (and `GetUserInfo`) return isn't consistent. Here is a truth table (note that I'm not sure if `IsZero` is actually defined on groups, but this to represent whether or not the group is a zero value) 

|                               | groupInfo.IsZero | errGroup != nil | group exists in MinIO |
| Get info a for a static group | 0                | 0               | 1                     |
| Get info a for a static group | 1                | 0               | 0                     |
| Get info a for a LDAP group   | 0                | 0               | 1                     |
| Get info a for a LDAP group   | 0                | 1               | 0                     |

Unfortunately, [AdminClient.GetGroupDescription](https://pkg.go.dev/github.com/minio/madmin-go/v2#AdminClient.GetGroupDescription) doesn't describe whether the expect return consistency, so I cannot tell if this is a bug or simply an expected behavior.

Note that exactly the same applies to `GetUserInfo`

I guess one issue of adding is condition, is that it could obfuscate other genuine issue when describing the LDAP group/user (e.g timeout, auth or API misbehavior). I will try and see if I can make this a little bit more resilient

Reply:
The reason for doing this is that `GetGroupDescription` (and `GetUserInfo`) expect the group or the user to be created before hand (e.g using a [minio_iam_group](https://registry.terraform.io/providers/aminueza/minio/latest/docs/resources/iam_group) or a [minio_iam_user](https://registry.terraform.io/providers/aminueza/minio/latest/docs/resources/iam_user)). However, when using a LDAP group or user, you must not create it, but instead should assign a policy right away.

To summarize:
- `GetGroupDescription` or `GetUserInfo` on a static group/user is expected to return a `200` with a list of policies attached to it:
  - If the list is including the policy at hand, it should do nothing (case `all good`, no error)
  - If the list isn't including the policy at hand, it should create (case `missing policy`, no error)
  - If the user/group doesn't exist, it aborts (case `mising user/group`, error)
- `GetGroupDescription` or `GetUserInfo` on a LDAP group/user is expected to return a `200` with a list of policies attached to it:
  - If the list is including the policy at hand, it should do nothing (case `all good`, no error)
  - If the user/group doesn't exist, it should create (case `mising user/group`, no error)

Hopefully, that makes sense now.

An example is provided in `examples/resources/minio_iam_group_policy_attachment/resource.tf` and `examples/resources/minio_iam_user_policy_attachment/resource.tf`

Reply:
I guess one probrem with this `isLDAP...` condition is that it would ignore any API error (e.g: timeout, auth, API misbehavior). I will try and see if I can make that is bit more resilient